File: backend/src/main/kotlin/network/bobnet/cms/model/content/Article.kt
package network.bobnet.cms.model.content

import network.bobnet.cms.model.data.RenderedArticle
import network.bobnet.cms.model.user.User
import java.time.LocalDateTime
import javax.persistence.*

@Entity
class Article(
        var title: String = "",
        var headline: String = "",
        @Column(columnDefinition = "LONGTEXT")
        var content: String = "",
        var featuredImage: String = "",
        @ManyToOne(cascade = [CascadeType.ALL]) var author: User = User(),
        var slug: String = "",
        var addedAt: LocalDateTime = LocalDateTime.now(),
        @Id @GeneratedValue var id: Long? = null,
        @ManyToMany var tags: MutableSet<Tag>? = null,
        @ManyToMany var categories: MutableSet<Category>? = null) {

    fun render() = id?.let {
        RenderedArticle(
                slug,
                title,
                headline,
                content,
                featuredImage,
                author,
                addedAt,
                tags,
                it
        )
    }
}

File: backend/src/entity/content/article.ts
// @ts-ignore
import { Schema, Model } from "mongoose";
import {Long} from "long";

module.exports = Model('Article', new Schema({
    title: String,
    headline: String,
    content: String,
    featuredImage: String,
    author: Object,
    slug: String,
    addedAt: Date,
    id: Long,
    tags: Object,
    categories: Object
}));